contracts/Staking.sol

```diff
@@ -640,6 +651,9 @@ contract Staking is Pauser, Whitelist {
 
         if (validator.status == dt.ValidatorStatus.Unbonded) {
             CELER_TOKEN.safeTransfer(delAddr, _tokens);
+            if (delegator.shares == 0) {
```

Comment:
Should we also remove the delegator from the delegator list when calling `completeUndelegate()`?

Reply:
The delegator may still have shares or already be removed when completeUndelegate happens which is after undeletation. Per UI design, the delegator won't show up as soon as no share for the validator. BTW, this will not prevent the delegator from completing undelegations anyway.

Reply:
So if the validator status is bonded, and then the delegator removes all its share by calling undelegateShares() and then calling completeUndelegate().  Since the delegator is not removed from the list, calling delegate() may add the delegate again (so the list contains two delegators with the same address).

Reply:
Sorry it seems like a mistake only remove delegator in Unbonded case.